### PR TITLE
Fix infinite loop from testing outside directory

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: python/hail/tests/utils.py [hunk 1]
```diff
@@ -19,8 +19,12 @@ def resource(filename):
     global _test_dir
     if _test_dir is None:
         path = '.'
+        i = 0
         while not os.path.exists(os.path.join(path, 'LICENSE')):
             path = os.path.join(path, '..')
+            i += 1
+            if i > 100:
```
Comment (jbloom22) [inline python/hail/tests/utils.py]: i prefer check of i == 100 when you're just incrementing by 1

Comment (tpoterba) [reply python/hail/tests/utils.py]: https://softwareengineering.stackexchange.com/questions/70996/versus-as-condition-in-a-for-loop

Comment (tpoterba) [reply python/hail/tests/utils.py]: the top answer here is one I like -- the greater than comparison indicates intent. They both do the same thing in this case, but I feel the comparison expresses the point of the code better than equality.

Comment (jbloom22) [reply python/hail/tests/utils.py]: yeah, that's reasonable

**Pull request merged**